hw/drivers/uart/max3107/syscfg.yml

```diff
@@ -0,0 +1,134 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+syscfg.defs:
+    MAX3107_0:
+        description: >
+            Create SPI RAM device instance.
+        value: 0
+
+    MAX3107_0_NAME:
+        description: >
+            MAX3107 device name.
+        value: '"max3107_0"'
+
+    MAX3107_0_UART_NAME:
+        description: >
+            MAX3107 UART device name.
+        value: '"uart4"'
+
+    MAX3107_0_SPI_BAUDRATE:
+        description: >
+            MAX3107 SPI clock speed in kHz.
+        value: 4000
+
+    MAX3107_0_UART_BAUDRATE:
+        description: >
+            MAX3107 UART clock speed in bits/second.
+        value: 115200
+
+    MAX3107_0_UART_DATA_BITS:
+        description: >
+            MAX3107 number of data bits.
+        range: 5,6,7,8
+        value: 8
+
+    MAX3107_0_UART_PARITY:
+        description: >
+            MAX3107 parity 0 - no parity, 1 - odd, 2 - even.
+        range: 0,1,2
+        value: 0
+
+    MAX3107_0_UART_STOP_BITS:
+        description: >
+            MAX3107 UART stop bits 1 or 2.
+        range: 1,2
+        value: 1
+
+    MAX3107_0_UART_FLOW_CONTROL:
+        description: >
+            MAX3107 UART flow control 0 - none, 1 - RTS CTS.
+        range: 0,1
+        value: 0
+
+    MAX3107_0_CS_PIN:
+        description: >
+            Chip select pin for MAX3107.
+        value: -1
+
+    MAX3107_0_SPI_BUS:
+        description: >
+            SPI interface bus name for MAX3107.
+        value : '"spi0"'
+
+    MAX3107_0_SPI_NUM:
+        description: >
+            SPI interface used for SPI RAM.
+        value: 0
+
+    MAX3107_0_OSC_FREQ:
+        description: >
+            External clock/oscillator frequency.
+        value:
```

Comment:
This value needs to be set to a default cause `clock_config` uses it, or you might need to handle the case where this is not set.

Reply:
There is not default value, it must be set according to actual external component value.

Reply:
Yeah, you are right since this is an external oscillator but there needs to be a range or something associated with this.

Reply:
Default value is not provided.
Range check is added to allow for valid frequencies 0.5-35 MHz.